cmd/kubeadm/app/cmd/phases/upgrade/node/controlplane.go

```diff
@@ -78,6 +79,12 @@ func runControlPlane() func(c workflow.RunData) error {
 			return errors.Wrap(err, "couldn't complete the static pod upgrade")
 		}
 
+		if features.Enabled(cfg.FeatureGates, features.UpgradeAddonsAfterControlPlane) {
```

Comment:
curious why this is gated... this is essentially a bugfix, right? the current order breaks https://kubernetes.io/releases/version-skew-policy/#kube-proxy

Reply:
context is here
https://github.com/kubernetes/kubernetes/pull/116570#issuecomment-1469578390

having it as a FG is more work for us and the users, but i personally had some concerns about the change and user assumptions WRT upgrade sequence.

perhaps if the e2e is green and the change is generally ok, we can still edit this relese note and remove the FG leaving it as a bugfix? one group of users it would break without a FG are those who interact with the addons right after `kubeadm apply` before upgrading more CP nodes. hopefully there is no such group..

@SataQiu and others, WDYT?

Reply:
having a way to opt out if there are issues maybe seems ok, but shouldn't this default on so kubeadm doesn't continue breaking supported skew by default?

Reply:
we could just use the FG mechanism and default it to true in alpha. it is a bit wierd but has been done for other k8s changes. then maybe we can just remove the FG in .29.

Reply:
another way we've approached changing undesirable/buggy behaviors is to gate them with a `Deprecated` feature gate, default it to false, and name it in a way that describes the unwanted behavior, so `UpgradeAddonsBeforeControlPlane`, state Deprecated, and default false, then mark the deprecated gate to be removed in a couple releases

Reply:
i see, it could be DeprecatedUpgradeAddonsAfterControlPlane

Reply:
It seems that we should update the feature gate in v1.28 to be deprecated and remove it in future releases.

Like below? `UpgradeAddonsAfterFirstControlPlane` may be better name.
```
UpgradeAddonsAfterFirstControlPlane: {FeatureSpec: featuregate.FeatureSpec{Default: false, PreRelease: featuregate.Deprecated}},
```

or like https://github.com/kubernetes/kubernetes/pull/86139, make it a GAed but not locked feature gate for users to switch. But that still needs `action-required` release note then.

```
UpgradeAddonsAfterControlPlane: {FeatureSpec: featuregate.FeatureSpec{Default: true, PreRelease: featuregate.GA}},
```

Reply:
right, i got confused.

so there is `PreRelease: featuregate.Deprecated`.

> UpgradeAddonsBeforeControlPlane, state Deprecated, and default false

we should do this.

Reply:
I opened https://github.com/kubernetes/kubernetes/pull/117660 to update it.

Then we should cleanup the release note in this PR.
/release-note-none